Framework/Crystal/src/SCDPanelErrors.cpp

```diff
@@ -1,711 +1,336 @@
   }
 
-  for (size_t i = EndX; i < nData; ++i)
-    out[i] = 0.;
-
-  g_log.debug() << "Parameters\n";
-
-  for (size_t i = 0; i < this->nParams(); ++i)
-    g_log.debug() << setw(20) << parameterName(i) << setw(20) << getParameter(i)
-                  << '\n';
-
-  g_log.debug() << "      chi Squared=" << std::setprecision(12) << chiSq
-                << '\n';
-
-  // Get values for test program. TODO eliminate
-  g_log.debug() << "  out[evenxx]=";
-  for (size_t i = 0; i < std::min<size_t>(nData, 30); ++i)
-    g_log.debug() << out[i] << "  ";
-
-  g_log.debug() << '\n';
+  Workspace_sptr ws = loadAlg->getProperty("OutputWorkspace");
+  API::Workspace_sptr resData =
+      boost::dynamic_pointer_cast<Mantid::API::Workspace>(ws);
+  loadWorkspace(resData);
 }
 
-double SCDPanelErrors::checkForNonsenseParameters() const {
-
-  double Dwdth = getParameter(0);
-  double Dhght = getParameter(1);
-  double x = getParameter(2);
-  double y = getParameter(3);
-  double z = getParameter(4);
-  double rx = getParameter(5);
-  double ry = getParameter(6);
-  double rz = getParameter(7);
-  double L0 = getParameter(8);
-  double T0 = getParameter(9);
-
-  double r = 0.;
-  if (L0 < 1.)
-    r = 1. - L0;
-
-  if (fabs(T0) > 20.)
-    r += (T0 - 20.) * 2.;
-
-  if (Dwdth < .5 || Dwdth > 2.)
-    r += 3. * fabs(1 - Dwdth);
-
-  if (Dhght < .5 || Dhght > 2.)
-    r += 3. * fabs(1. - Dhght);
-
-  if (fabs(x) > .35)
-    r += fabs(x) * .2;
-
-  if (fabs(y) > .35)
-    r += fabs(y) * .2;
-
-  if (fabs(z) > .35)
-    r += fabs(z) * .2;
-
-  if (fabs(rx) > 15.)
-    r += fabs(rx) * .02;
-
-  if (fabs(ry) > 15.)
-    r += fabs(ry) * .02;
-
-  if (fabs(rz) > 15.)
-    r += fabs(rz) * .02;
-
-  return 5. * r;
+/**
+ * Load the points from a PeaksWorkspace
+ * @param wsName :: The workspace to load from
+ */
+void SCDPanelErrors::loadWorkspace(const std::string &wsName) const {
+  auto ws = AnalysisDataService::Instance().retrieveWS<API::Workspace>(wsName);
+  loadWorkspace(ws);
 }
 
-void updateDerivResult(PeaksWorkspace_sptr peaks, V3D &unRotDeriv,
-                       Matrix<double> &Result, size_t peak,
-                       vector<int> &peakIndx) {
-  Matrix<double> GonMatrix =
-      peaks->getPeak(peakIndx[peak]).getGoniometerMatrix();
-  GonMatrix.Invert();
-
-  V3D RotDeriv = GonMatrix * unRotDeriv;
-
-  for (int kk = 0; kk < 3; ++kk)
-    Result[kk][peak] = RotDeriv[kk];
+/**
+ * Load the points from a PeaksWorkspace
+ * @param ws :: The workspace to load from
+ */
+void SCDPanelErrors::loadWorkspace(boost::shared_ptr<API::Workspace> ws) const {
+  m_workspace = ws;
+  m_setupFinished = false;
 }
 
-void SCDPanelErrors::functionDeriv1D(Jacobian *out, const double *xValues,
-                                     const size_t nData) {
-  if (nData <= 0)
+/**
+  * Fill in the workspace name and bank
+  */
+void SCDPanelErrors::setupData() const {
+  if (m_setupFinished) {
+    g_log.debug() << "Re-setting isn't required.";
```

Comment:
Since you don't need the streams interface, `g_log("Re-setting isn't required.");` is preferred.